test/core/transport/chttp2/http2_client_transport_test.cc

```diff
@@ -0,0 +1,61 @@
+//
+//
+// Copyright 2024 gRPC authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+//
+
+#include "src/core/ext/transport/chttp2/transport/http2_client_transport.h"
+
+#include "absl/log/check.h"
+#include "absl/log/log.h"
+#include "gtest/gtest.h"
+#include "src/core/lib/event_engine/default_event_engine.h"
+#include "test/core/transport/chaotic_good/mock_promise_endpoint.h"
+#include "test/core/transport/chaotic_good/transport_test.h"
+
+using grpc_core::chaotic_good::testing::MockPromiseEndpoint;
+using grpc_event_engine::experimental::EventEngine;
+
+namespace grpc_core {
+namespace http2 {
+namespace testing {
+
+TEST(Http2ClientTransportTest, TestHttp2ClientTransportObjectCreation) {
+  MockPromiseEndpoint control_endpoint(1);
```

Comment:
Borrowed these mock classes from Chaotic Good. I hope that is ok.

Reply:
Yeah it is, but maybe let's move them to `test/core/transport/util/mock_promise_endpoint.h`

Reply:
Done